File: frontend/src/api/keyApplication.ts
import request from './request'

export interface KeyApplication {
  id?: number
  userId: number
  tenantId?: string
  appId?: string
  keyName: string
  allowedModels?: string
  reason?: string
  approvalStatus?: number
  approverId?: number
  approvalComment?: string
  approvalTime?: string
  createTime?: string
}

export const submitKeyApplication = (data: Omit<KeyApplication, 'id' | 'approvalStatus'>) => {
  return request.post<any, KeyApplication>('/api/admin/key-applications', data)
}

export const getMyKeyApplications = (userId: number) => {
  return request.get<any, KeyApplication[]>('/api/admin/key-applications/my', {
    params: { userId },
  })
}

export const getAllKeyApplications = (status?: number) => {
  return request.get<any, KeyApplication[]>('/api/admin/key-applications', {
    params: { status },
  })
}

export const approveKeyApplication = (id: number, approverId: number, comment?: string) => {
  return request.post<any, any>(`/api/admin/key-applications/${id}/approve`, null, {
    params: { approverId, comment },
  })
}

export const rejectKeyApplication = (id: number, approverId: number, comment?: string) => {
  return request.post<any, boolean>(`/api/admin/key-applications/${id}/reject`, null, {
    params: { approverId, comment },
  })
}

File: gateway-admin/src/main/java/com/aigateway/admin/entity/KeyApplication.java
package com.aigateway.admin.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * API Key 申请实体
 */
@Data
@TableName("key_application")
public class KeyApplication {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long userId;
    
    private String tenantId;
    
    private String appId;
    
    private String keyName;
    
    private String allowedModels;
    
    private String reason;
    
    private Integer approvalStatus;
    
    private Long approverId;
    
    private String approvalComment;
    
    private LocalDateTime approvalTime;
    
    private LocalDateTime createTime;
    
    private LocalDateTime updateTime;
}
